app/service/rest_svc.py

```diff
@@ -40,6 +41,32 @@ def __init__(self):
             await self.get_service('data_svc').load_data(d)
         return await self._poll_for_data('adversaries', dict(adversary_id=i))
 
+    async def update_planner(self, data):
+        """
+        Update a new planner from either the GUI or REST API with new stopping conditions.
+        This overwrites the existing YML file.
+        :param data:
+        :return: the ID of the created adversary
+        """
+        planner = (await self.get_service('data_svc').locate('planners', dict(name=data['name'])))[0]
+        id = planner.id
+        _, file_path = await self.get_service('file_svc').find_file_path('%s.yml' % id, location='data')
+        if not file_path:
+            file_path = 'data/planners/%s.yml' % id
+        with open(file_path, 'r') as f:
+            planner_obj = yaml.load(f.read(), Loader=yaml.FullLoader)
+        sc = data.get('stopping_conditions')
+        if sc:
+            sc = [{s.get('trait'): s.get('value')} for s in sc]
+        planner_obj['stopping_conditions'] = sc
+        with open(file_path, 'w') as f:
+            f.write(yaml.dump(planner_obj))
+        planner.stopping_conditions = [Fact(trait=f.get('trait'), value=f.get('value'))
+                                       for f in data['stopping_conditions']]
+        await self.get_service('data_svc').remove('planners', dict(name=data['name']))
```

Comment:
we shouldn't have to remove the planner to store it, we should update instead. look at the c_agent.py for a nice example of how the store function can create or update an object.